lib/EnzymeCore/test/runtests.jl

```diff
@@ -0,0 +1,21 @@
+using Test
+using EnzymeCore
+
+import EnzymeCore.EnzymeRules: forward, has_frule_from_sig
+
+g(x) = x ^ 2
+function forward(::Const{typeof(g)}, ::Type{<:Const}, x::Const)
+    return Const(g(x.val))
+end
+
+@test has_frule_from_sig(Base.signature_type(g, Tuple{Float64}))
+
+f(;kwargs) = 1.0
+
+function forward(::Const{typeof(f)}, ::Type{<:Const}; kwargs...)
+    return Const(f(; kwargs...))
+end
+
+@test has_frule_from_sig(Base.signature_type(f, Tuple{}))
```

Comment:
Is it useful to test where we have a kwarg function but a rule that doesn't take the kwarg?

Reply:
Yeah we will error for Enzyme proper (at least that's the goal)